pkg/sql/distsql_physical_planner.go

```diff
@@ -1269,7 +1267,7 @@ func (h *distSQLNodeHealth) checkSystem(
 		// artifact of rpcContext's reconnection mechanism at the time of
 		// writing). This is better than having it used in 100% of cases
 		// (until the liveness check below kicks in).
-		if err := h.connHealthSystem(roachpb.NodeID(sqlInstanceID), rpc.DefaultClass); err != nil {
+		if err := h.connHealthWithResolve(sqlInstanceID); err != nil {
```

Comment:
As a follow-up we can explore if we can eleminate `checkSystem` function and just use `checkInstanceHealth` (then we wouldn't be needing `conHealthWithResolve`).